### PR TITLE
Refine PADDLE_ENFORCE

### PR DESCRIPTION
**Pull request by GaoWei8**
### PR types
Function optimization

### PR changes
OPs 

### Describe
Refine PADDLE_ENFORCE of paddle/fluid/operators/jit/    paddle/fluid/operators/benchmark

### CODE REVIEW

File: paddle/fluid/operators/jit/gen_base.cc [hunk 1]
```diff
@@ -49,9 +49,12 @@ void GenBase::dumpCode(const unsigned char* code) const {
 void* GenBase::operator new(size_t size) {
   void* ptr;
   constexpr size_t alignment = 32ul;
-  PADDLE_ENFORCE_EQ(posix_memalign(&ptr, alignment, size), 0,
-                    "GenBase Alloc %ld error!", size);
-  PADDLE_ENFORCE(ptr, "Fail to allocate GenBase CPU memory: size = %d .", size);
+  PADDLE_ENFORCE_EQ(
+      posix_memalign(&ptr, alignment, size), 0,
+      platform::errors::InvalidArgument("GenBase Alloc %ld error!", size));
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/gen_base.cc]: what is `GenBase`, can we use an understandable description here?

File: paddle/fluid/operators/jit/gen_base.cc [hunk 2]
```diff
@@ -49,9 +49,12 @@ void GenBase::dumpCode(const unsigned char* code) const {
...
+  PADDLE_ENFORCE_NOT_NULL(
+      ptr, platform::errors::InvalidArgument(
+               "Fail to allocate GenBase CPU memory: size = %d .", size));
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/gen_base.cc]: same above

File: paddle/fluid/operators/jit/helper.cc [hunk 3]
```diff
@@ -66,7 +66,8 @@ const char* to_string(KernelType kt) {
     ONE_CASE(kEmbSeqPool);
     ONE_CASE(kSgd);
     default:
-      PADDLE_THROW("Not support type: %d, or forget to add it.", kt);
+      PADDLE_THROW(platform::errors::Unimplemented(
+          "Not support type: %d, or forget to add it.", kt));
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/helper.cc]: type of what?   is this user `forget`? if user can not `add` here, I think we can only tell users `Upsupported`

File: paddle/fluid/operators/jit/helper.cc [hunk 4]
```diff
@@ -79,7 +80,8 @@ const char* to_string(SeqPoolType tp) {
     ONE_CASE(kAvg);
     ONE_CASE(kSqrt);
     default:
-      PADDLE_THROW("Not support type: %d, or forget to add it.", tp);
+      PADDLE_THROW(platform::errors::Unimplemented(
+          "Not support type: %d, or forget to add it.", tp));
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/helper.cc]: same above

File: paddle/fluid/operators/jit/helper.cc [hunk 5]
```diff
@@ -100,7 +102,8 @@ KernelType to_kerneltype(const std::string& act) {
   } else if (lower == "tanh" || lower == "vtanh") {
     return kVTanh;
   }
-  PADDLE_THROW("Not support type: %s, or forget to add this case", act);
+  PADDLE_THROW(platform::errors::Unimplemented(
+      "Not support type: %s, or forget to add this case", act));
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/helper.cc]: same above

File: paddle/fluid/operators/jit/helper.cc [hunk 6]
```diff
@@ -109,12 +112,18 @@ void pack_weights<float>(const float* src, float* dst, int n, int k) {
   int block, rest;
   const auto groups = packed_groups(n, k, &block, &rest);
   std::for_each(groups.begin(), groups.end(), [&](int i) {
-    PADDLE_ENFORCE_GT(i, 0, "each element of groups should be larger than 0.");
+    PADDLE_ENFORCE_GT(i, 0, platform::errors::InvalidArgument(
+                                "Each element of groups should be larger than "
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/helper.cc:116]: here I don't understand, if `i < 0`, such as -1, `the -1th element`?

File: paddle/fluid/operators/jit/helper.cc [hunk 7]
```diff
@@ -109,12 +112,18 @@ void pack_weights<float>(const float* src, float* dst, int n, int k) {
...
+                                "0. However the %dth element doesn't satify.",
+                                i));
   });
   int sum = std::accumulate(groups.begin(), groups.end(), 0);
   std::memset(dst, 0, k * sum * block * sizeof(float));
   PADDLE_ENFORCE_GE(sum * block, n,
-                    "The packed n should be equal to or larger than n");
+                    platform::errors::InvalidArgument(
+                        "The packed n should be equal to or larger than n. "
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/helper.cc]: n is what? maybe users don't know `n`?

File: paddle/fluid/operators/jit/helper.cc [hunk 8]
```diff
@@ -136,7 +145,8 @@ void pack_weights<float>(const float* src, float* dst, int n, int k) {
 template <typename T>
 typename std::enable_if<!std::is_same<T, float>::value>::type pack_weights(
     const T* src, T* dst, int n, int k) {
-  PADDLE_THROW("Only support pack with float type.");
+  PADDLE_THROW(
+      platform::errors::InvalidArgument("Only support pack with float type."));
```
Comment (chenwhql) [inline paddle/fluid/operators/jit/helper.cc]: support -> supports?

File: paddle/fluid/operators/jit/gen_base.cc (hunk 1 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/gen_base.cc]: replace with Jitcode generator

File: paddle/fluid/operators/jit/gen_base.cc (hunk 2 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/gen_base.cc]: Done.

File: paddle/fluid/operators/jit/helper.cc (hunk 3 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/helper.cc]: Done

File: paddle/fluid/operators/jit/helper.cc (hunk 4 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/helper.cc]: Done

File: paddle/fluid/operators/jit/helper.cc (hunk 5 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/helper.cc]: Done.

File: paddle/fluid/operators/jit/helper.cc (hunk 6 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/helper.cc:116]: replace with the accurate value

File: paddle/fluid/operators/jit/helper.cc (hunk 7 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/helper.cc]: explian the meaning of n

File: paddle/fluid/operators/jit/helper.cc (hunk 8 above)
Comment (GaoWei8) [reply paddle/fluid/operators/jit/helper.cc]: Done

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**